cpp/daal/include/data_management/data/internal/numeric_table_sycl_csr.h

```diff
@@ -0,0 +1,673 @@
+
+    services::Status allocateDataMemoryImpl(daal::MemType /*type*/ = daal::dram) DAAL_C11_OVERRIDE
+    {
+        return services::Status(services::ErrorMethodNotSupported);
+    }
+
+    void freeDataMemoryImpl() DAAL_C11_OVERRIDE
+    {
+        _values = services::internal::sycl::UniversalBuffer();
+        _colIndices.reset();
+        _rowOffsets.reset();
+        _memStatus = notAllocated;
+    }
+
+    /** \private */
+    template <typename Archive, bool onDeserialize>
+    services::Status serialImpl(Archive * archive)
+    {
+        services::Status status = SyclNumericTable::serialImpl<Archive, onDeserialize>(archive);
+
+        size_t dataSize = 0;
+        if (!onDeserialize)
+        {
+            dataSize = getDataSize();
+        }
+        archive->set(dataSize);
+
+        if (onDeserialize)
+        {
+            if (isCpuTable())
+            {
+                DAAL_CHECK_STATUS(status, _cpuTable->allocateDataMemory(dataSize));
+            }
+            else
+            {
+                DAAL_CHECK_STATUS(status, allocateDataMemory(dataSize));
+            }
+        }
+
+        size_t nfeat = getNumberOfColumns();
+        size_t nobs  = getNumberOfRows();
+
+        if (nfeat > 0)
+        {
+            NumericTableFeature & f = (*_ddict)[0];
+            if (isCpuTable())
+            {
+                char * data         = NULL;
+                size_t * colIndices = NULL;
+                size_t * rowOffsets = NULL;
+
+                _cpuTable->getArrays(&data, &colIndices, &rowOffsets);
+                archive->set(data, dataSize * f.typeSize);
+                archive->set(colIndices, dataSize);
+                archive->set(rowOffsets, nobs + 1);
+            }
+            else
+            {
+                services::SharedPtr<size_t> hostColIndices = _colIndices.toHost(data_management::readOnly, status);
+                DAAL_CHECK_STATUS_VAR(status);
+                services::SharedPtr<size_t> hostRowOffsets = _rowOffsets.toHost(data_management::readOnly, status);
+                DAAL_CHECK_STATUS_VAR(status);
+
+                switch (f.indexType)
+                {
+                case features::DAAL_FLOAT32:
+                {
+                    services::SharedPtr<float> hostData = _values.get<float>().toHost(data_management::readOnly, status);
+                    DAAL_CHECK_STATUS_VAR(status);
+                    archive->set(hostData.get(), dataSize);
+
+                    break;
+                }
+                case features::DAAL_FLOAT64:
+                {
+                    services::SharedPtr<double> hostData = _values.get<double>().toHost(data_management::readOnly, status);
+                    DAAL_CHECK_STATUS_VAR(status);
+                    archive->set(hostData.get(), dataSize);
+
+                    break;
+                }
+
+                default: status = services::Status(services::ErrorIncorrectParameter);
+                }
```

Comment:
Can you use `BufferHostInterpreter` as in SyclSoaTable?
https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/include/data_management/data/internal/numeric_table_sycl_soa.h#L427

Reply:
Good idea, thanks! Done.